README.md

```diff
@@ -1,78 +1,405 @@
+If you take a look at the generated file `1.csv` you'll see that data for the `firstName` field does not objy the given constraints, whilst those for `age` and `nationalInsurance` are correct:
 
-| Product | Description | Status | Notes |
-| ---- | ---- | ---- | ---- |
-| Profile | A representation of the data schema and rules which can be used to generate data | Stable | |
-| Generator | A tool for generating data from a given profile | Stable - beta | Supports data generation and generation of data that has been [delibrately violated](./generator/docs/DeliberateViolation.md). |
+~~~
+firstName,age,nationalInsurance
+-619248029,71,"HT849919"
+08-12-2001 02:53:16,15,
+"Lorem Ipsum",11,
+"Lorem Ipsum",71,"WX004081"
+1263483797,19,"HG054666"
+,75,"GE023082"
+"Lorem Ipsum",59,"ZM850737C"
+[...]
+~~~
 
-See [overview](./docs/Profiles.md) and [schema](./docs/Schema.md) for details about profiles and how they work.
+However, it might be a surprise to see nulls, numbers and dates as values for the `firstName` field alongside strings that do not match the regex given in the profile. This is because these are all defined as a single rule within the profile. You have a couple of options if you want to ensure that `firstName` is not null and a string, the first is to inform the generator that it should not violate specific constraint types:
 
-* [Generator](./generator/README.md)
-  * [Generation algorithm](./generator/docs/GenerationAlgorithm.md)
-  * [Set data generation process](./generator/docs/SetRestrictionAndGeneration.md)
-  * [Decision trees](./docs/DecisionTrees/DecisionTrees.md)
+~~~
+$ java -jar generator.jar generate --violate --dont-violate=ofType null \
+  --max-rows=100 --overwrite profile.json out
+~~~
 
-* [Frequently asked questions](docs/FrequentlyAskedQuestions.md)
+Or, alternatively, you can re-arrange your constraints so that the ones that define types / null, are grouped as a single rule. After By re-grouping constraints, the following output, with random strings that violate the regex constraint, is generated:
 
-## Contributing
+firstName,age,nationalInsurance
+"",43,"PT530853D"
+"&#48583;",56,"GE797875M"
+"&#37030;&#29187;",84,"JA172890M"
+"J&#14339;&#25031;&#22289;&#12961;&#20459;&#26440;",32,"AE613401F"
+"&#20469;&#53921;",38,"TS256211F"
+"M",60,"GE987171M"
+"M",7,
+"M&#42894;&#4741;&#59414;&#21101;&#42596;&#21708;&#53662;&#44242;",97,"EN082475C"
+")",80,"BX025130C"
+",&#9281;&#51869;",60,"RW177969"
+"5&#6275;&#54430;&#65199;&#17599;&#22267;",57,"RY904705"
+[...]
 
-See [contributing](./.github/CONTRIBUTING.md) guidance and the [key decisions log](docs/KeyDecisions.md).
+**NOTE** - the above is not formatted as a code block because GitHub flavor markdown doesn't support the unicode character set which th generator outputs. On a related note, we are considering adding a feature that allows you to [specify / restrict the character set](https://github.com/ScottLogic/datahelix/issues/294) in a future release.
```

Comment:
If you use ``` instead of ~~~ it might resolve the issue?

![image](https://user-images.githubusercontent.com/44162572/54591421-1a2cd500-4a22-11e9-8c2e-f4cf55134eb9.png)

Reply:
Ah, missed the point here.

How about this?

```
firstName,age,nationalInsurance
"�",43,"PT530853D"
"뷇",56,"GE797875M"
"邦爃",84,"JA172890M"
"J㠃懇圑㊡俫杈",32,"AE613401F"
"俵튡",38,"TS256211F"
"M",60,"GE987171M"
"M",7,
"Mꞎኅ剭Ꙥ哌톞곒",97,"EN082475C"
")",80,"BX025130C"
",⑁쪝",60,"RW177969"
"5ᢃ풞ﺯ䒿囻",57,"RY904705" 
[...]
```

Reply:
@ColinEberhardt I've just pushed to your branch, i think this fixes the output/code-block issue

Reply:
> I've just pushed to your branch, i think this fixes the output/code-block issue

Cool - thanks for fixing that. Although I'd still like to maintain the note about character sets that referenced the relevant issue. It is a good demonstration of some of the things we are thinking about post-beta.

Reply:
Note reintroduced.